### PR TITLE
Milestone 3: Feature | associations setup

### PR DESCRIPTION
**Pull request by ITurres**
# 🚩I have made the following changes to complete OOP School Library - Milestone/Project 3

### Type of **`files`** added / modified

My Skills

### In this new Branch, I have **➕ADDED➕** the succeeding **\*📂 folders** and/or **\*\*📄 files** to complete this milestone's [requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_associations.md)

# \*📂 root/

- ## \*📂 lib/

  - **\*\*📄 book.rb** ➕
  - **\*\*📄 rental.rb** ➕
  - **\*\*📄 classroom.rb** ➕

### I have **✏️MODIFIED✏️** the succeeding **\*📂 folders** and/or **📄\*\* files**

# \*📂 root/

- ## \*📂 lib/

  - **\*\*📄 person.rb** ✏️

    - The **`-Person-`** class was modified to be part of the `many-to-many` and `has-many-through` associations with the **`-Rental-`** and **`-Book-`** classes.

  - **\*\*📄 student.rb** ✏️

    - I had establish bidirectional association: Student adds to **`-Classroom-`**,
      and **`-Classroom-`** assigns to Student.

---

### Thank you for taking the time to review this PR ⭐

---

If you require additional information or have any questions, don't hesitate to get in touch with me on Slack as _**Arturo (Arthur) Emanuel Guerra Iturres**_. I'll be happy to help you out. 🎯

---

### CODE REVIEW

File: lib/student.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@ class Student < Person
   def initialize(age, classroom, name = 'Unknown', parent_permission: true)
     super(age, name, parent_permission)
     @classroom = classroom
+    classroom&.add_student(self)
```
Comment (abdulhamiid) [inline lib/student.rb:9]: According to the project requirements, please, create `a method` that serves as the 'belongs-to' side (a student belongs to a classroom). 

image

So any time we add a student to a classroom, the classroom of the student is set as well, and vice versa (Both ways relationships). 💪🛠

Link to the [article with examples](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @ITurres 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! :running_man: :runner: 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Descriptions are very descriptive
- ✔️ Set up associations between classes and objects.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: lib/student.rb (hunk 1 above)
Comment (ITurres) [reply lib/student.rb:9]: Hi @abdulhamiid, Thank you for your diligent review of this PR! Your insights are much appreciated. Let's dive into the changes suggested! 🎯 🙏🏻

The 
```rb
 classroom&.add_student(self)
```
makes use of the **`&.`** operator for safe navigation, allowing to call a method on an object if the object is not **_nil_**. In this case, it's calling the `'add_student'` method on the classroom object if **`-ClassRoom-`** is not **_nil_**.
This already handles the relationship between **`-Student-`** and **`-ClassRoom-`** by using the `'add_student'` method within the **`-ClassRoom-`** class.
The code establishes a `belongs-to` relationship. So when a new Student is created and added to a **`-ClassRoom-`**, the student's classroom attribute is set to the corresponding classroom, and the student is added to the classroom's list of students. This indicates that the **`-Student-`** class belongs to a **`-ClassRoom-`** class.

I hope this is of help! 🤠

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @ITurres , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Excellent communication with reviewer

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ITurres** [comment]
Hello @t-yanick Tazoh! 🎉 Thank you for approving the PR! 👍 I wanted to make sure that the implementation mentioned above is clear and aligns with the intended changes. If you or _Adio_ have any insights or feedback, I'm all ears. Looking forward to your thoughts! Cheers!

**Pull request merged**